### ISSUE TITLE
Cache bad ES queries

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#21012]
In large running systems, we see a lot of errors in the logs like these - where dotCMS or the customer is posting a bad query to elasticsearch.  If this bad query is being generated on a page or template under load, it hits ES repeatedly and floods the logs with the same messages over and over.  

We already have an ES query cache.  Let's just cache an empty result for bad queries and not hit ES.

```
[16/09/21 23:42:11:813 AEST]  WARN business.ESContentFactoryImpl: Elasticsearch error in index 'cluster_prod-firstmac-5382.live_20210910212446'
[16/09/21 23:42:11:813 AEST]  WARN business.ESContentFactoryImpl: ES Query: {"size":10000,"timeout":"15000ms","query":{"match_all":{"boost":1.0}},"post_filter":{"query_string":{"query":"+identifier: +live:true +deleted:false +((+owner:anonymous +ownercanread:true) (permissions:p654b0931-1027-41f7-ad4d-173115ed8ec1.1p* permissions:p999cd6bf-5cef-4729-8543-696086143884.1p* permissions:pedecd377-2321-4803-aa8b-89797dd0d61f.1p* ) (permissions:p654b0931-1027-41f7-ad4d-173115ed8ec1.1p*) (permissions:p999cd6bf-5cef-4729-8543-696086143884.1p*)) +live:true","fields":[],"type":"best_fields","default_operator":"or","max_determinized_states":10000,"enable_position_increments":true,"fuzziness":"AUTO","fuzzy_prefix_length":0,"fuzzy_max_expansions":50,"phrase_slop":0,"escape":false,"auto_generate_synonyms_phrase_query":true,"fuzzy_transpositions":true,"boost":1.0}},"_source":{"includes":["inode","identifier"],"excludes":[]},"sort":[{"moddate_dotraw":{"order":"asc"}}]}
[16/09/21 23:42:11:813 AEST]  WARN business.ESContentFactoryImpl: Class org.elasticsearch.ElasticsearchStatusException: Elasticsearch exception [type=parse_exception, reason=parse_exception: Encountered " "+" "+ "" at line 1, column 13.
Was expecting one of:
    <BAREOPER> ...
    "(" ...
    "*" ...
    <QUOTED> ...
    <TERM> ...
    <PREFIXTERM> ...
    <WILDTERM> ...
    <REGEXPTERM> ...
    "[" ...
    "{" ...
    <NUMBER> ...
    ]
[16/09/21 23:42:11:813 AEST]  WARN business.ESContentFactoryImpl: ----------------------------------------------
```

**Comment by wezell** [comment]
PR: #21013

**Comment by wezell** [comment]
Note to Internal QA - we should test that we are not caching an empty result when elasticsearch is not reachable

**Comment by wezell** [comment]
An `ElasticsearchStatusException` causes us to cache a bad query:
```
14:58:13.838  WARN  business.ESContentFactoryImpl - Elasticsearch error in index 'cluster_b2a55a0e47.live_20210921090328'
14:58:13.838  WARN  business.ESContentFactoryImpl - Thread: url:POST//demo.dotcms.com/dwr/call/plaincall/BrowserAjax.getHostsIncludeAll.dwr | lang:1 | ip:127.0.0.1 | Admin:true | start:09-21-2021 01:40:04 EDT  ref:https://local.dotcms.site:8443/c/portal/layout?p_l_id=71b8a1ca-37b6-4b6e-a43b-c7482f28db6c&p_p_id=c_Call-To-Action&p_p_action=0&&dm_rlout=1&r=1632248890230&in_frame=true&frame=detailFrame&container=true&angularCurrentPortlet=c_Call-To-Action
14:58:13.838  WARN  business.ESContentFactoryImpl - ES Query: {"size":10000,"timeout":"15000ms","query":{"match_all":{"boost":1.0}},"post_filter":{"query_string":{"query":"+contenttype:languagevariable +languagevariable.key_dotraw:\"tag-system-host\" +languageid:1 +live:true +deleted:false","fields":[],"type":"best_fields","default_operator":"or","max_determinized_states":10000,"enable_position_increments":true,"fuzziness":"AUTO","fuzzy_prefix_length":0,"fuzzy_max_expansions":50,"phrase_slop":0,"escape":false,"auto_generate_synonyms_phrase_query":true,"fuzzy_transpositions":true,"boost":1.0}},"_source":{"includes":["inode","identifier"],"excludes":[]},"sort":[{"moddate":{"order":"desc"}}],"track_total_hits":10000000}
14:58:13.838  WARN  business.ESContentFactoryImpl - Class org.elasticsearch.ElasticsearchStatusException: method [POST], host [https://oboxturbo:9200], URI [/cluster_b2a55a0e47.live_20210921090328/_search?typed_keys=true&ignore_unavailable=false&expand_wildcards=open&allow_no_indices=true&ignore_throttled=true&search_type=query_then_fetch&batched_reduce_size=512&ccs_minimize_roundtrips=true], status line [HTTP/1.1 503 Service Unavailable]
Open Distro Security not initialized.
```

**Comment by bryanboza** [comment]
Fixed, Tested on release-21.10 // Docker
